### ISSUE TITLE
Multiple TLS connections

### ISSUE THREAD

**Issue by nmihaica** [axios/axios#2473]
**Describe the issue**
Pondering over ways to achieve multiple TLS connections over same agent.
```
client -> server_a -> server_b

client:
iniates request through some front-end
which then calls action
to the server_a.

server_a:
receives, properly examine
which client sent the request
initiate request to the server_b 
get and set appropriate certificates
sign the document

server_b:
(not in domain of my solution, third-party)
recieves request from server_a,
validate and verify and return response to the 
server_a with verified data (server_a then responds back to the
client originally finishing the main request)
```

**Explanation**

I will manage x509 certificates inside database. Is this naive? What am I missing?

**Example Code**
```
let api = axios.create({
    baseURL: opt.demo ? 'demo.server:8400 : 'prod.server:8400',
    httpsAgent: new https.Agent({
      rejectUnauthorized: true,
      ca: fs.readFileSync('./certs/ca_bundle.pem'),
      key: changeKeyInplaceSomehow(),
      cert: changeCertInplaceSomehow(),
    })
  })
```

**Conclusion**
So at the end that you can call the same service, but it will run with different key and cert for the different user
```js
//Will user different key/cert pair for different user
api.post('/service')
.then(response => res)
.catch(err => err)
```

**Comment by yasuf** [comment]
do you need to handle the agent's key and cert on the client? can that be within server_a ?

**Comment by nmihaica** [comment]
There is option to use agent on the client (which then would mean two seperates connections)
```
client -> server_b // send signed document, recieve answer
client -> server_a // extract answer and save new object in database
```
Additionally one should wait for response before initiating , but this solution has its advantages. However I intended it to be on server_a. I am still early and can change it to client. Whatever solution I would like to know if something like this is possible. Considering the move to the client I would bypass this problem and probably could have cleaner solution, however there is new one. How to setup certificates properly inside browser Javascript?

```
users have 1:1 connection with certificates
1 end user - 1 certificate pair(key + cert)
```

**Disclaimer**
While not axios issue *per se*, more of an architectural one I opened it so that it could be discussed because potential discussion/solution could be beneficial for later usage as axios is fairly used both on front aswell as back-end.
